### PR TITLE
add txpool validator

### PR DESCRIPTION
**Pull request by HaoXuan40404**

### CODE REVIEW

**Review by morebtcg — DISMISSED** [review]

File: bcos-protocol/bcos-protocol/TransactionStatus.h [hunk 1]
```diff
@@ -64,6 +64,10 @@ enum class TransactionStatus : int32_t
     RequestNotBelongToTheGroup = 10009,
     TransactionPoolTimeout = 10010,
     AlreadyInTxPoolAndAccept = 10011,
+    OverFlowValue = 10012,
+    OversizedData = 10013,
+    NoEOAAccount = 10014,
```
Comment (bxq2011hust) [inline bcos-protocol/bcos-protocol/TransactionStatus.h:69]:
```suggestion
    NotEOAAccount = 10014,
```

File: bcos-framework/bcos-framework/txpool/Constant.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ */
+#pragma once
+#include <cstddef>
+
+// u256 with 0x prefix
+const size_t TRANSACTION_VALUE_MAX_LENGTH = 256 * 2 + 2;
+// EIP-170
+const int MAX_CODE_SIZE = 0x6000;
+// EIP 3860
+const int MAX_INITCODE_SIZE = 2 * MAX_CODE_SIZE;
```
Comment (bxq2011hust) [inline bcos-framework/bcos-framework/txpool/Constant.h:28]: The max init code size maybe should be bigger

File: bcos-rpc/bcos-rpc/validator/RpcValidator.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+//     return chainId == _chainId;
+// }
+
+// bool RpcValidator::checkBcosReplayedProtected(
+//     protocol::Transaction::Ptr _tx, std::string_view _chainId)
+// {
+//     return _tx->chainId() == _chainId;
+// }
+
+task::Task<protocol::TransactionStatus> RpcValidator::checkTransactionValidator(
```
Comment (bxq2011hust) [inline bcos-rpc/bcos-rpc/validator/RpcValidator.cpp:50]:
```suggestion
task::Task<protocol::TransactionStatus> RpcValidator::validateTransaction(
```

File: bcos-protocol/bcos-protocol/TransactionStatus.h [hunk 4]
```diff
@@ -64,6 +64,10 @@ enum class TransactionStatus : int32_t
     RequestNotBelongToTheGroup = 10009,
     TransactionPoolTimeout = 10010,
     AlreadyInTxPoolAndAccept = 10011,
+    OverFlowValue = 10012,
+    OversizedData = 10013,
```
Comment (bxq2011hust) [inline bcos-protocol/bcos-protocol/TransactionStatus.h:68]: InitCodeOversized

File: bcos-rpc/bcos-rpc/web3jsonrpc/endpoints/EthEndpoint.cpp [hunk 5]
```diff
@@ -429,10 +442,31 @@ task::Task<void> EthEndpoint::sendRawTransaction(const Json::Value& request, Jso
...
+    if (!config.has_value())
+    {
+        BOOST_THROW_EXCEPTION(
+            JsonRpcException(JsonRpcError::InvalidParams, "ChainId not available!"));
+    }
+    auto [chainId, _] = config.value();
+    auto txChainId = removeHexPrefix(toQuantity(web3Tx.chainId.value_or(0)));
+    if (txChainId != chainId)
+    {
+        std::cout << "txChainId: " << txChainId << ", chainId: " << chainId << std::endl;
```
Comment (bxq2011hust) [inline bcos-rpc/bcos-rpc/web3jsonrpc/endpoints/EthEndpoint.cpp:456]: use log

**Review by bxq2011hust — APPROVED** [review]

**Pull request merged**